tensorflow/python/util/nest.py

```diff
@@ -128,7 +137,17 @@ def _sequence_like(instance, args):
   Returns:
     `args` with the type of `instance`.
   """
-  if _is_mapping(instance):
+  if _is_mutable_mapping(instance):
+    result = dict(zip(_sorted(instance), args))
+    instance_type = type(instance)
+    if instance_type == _collections.OrderedDict:
+      d = _collections.OrderedDict(instance.default_factory)
+      for key in instance:
+        d[key] = result[key]
+      return d
+    else:
+      return instance_type((key, result[key]) for key in instance)
+  elif _is_mapping(instance):
```

Comment:
`defaultdict` should be captured by the branch above, so we should no longer need the special case below.

Reply:
The new commit has the changes you requested.